### PR TITLE
update_state_sync if requested state + timeout message fix

### PR DESCRIPTION
**Pull request by Kouprin**
move update_sync_status under request_shard() + fix timeout message

### CODE REVIEW

File: chain/client/src/sync.rs [hunk 1]
```diff
@@ -522,7 +522,7 @@ impl StateSync {
                         let prev = shard_sync_download.downloads[0].prev_update_time;
                         let error = shard_sync_download.downloads[0].error;
                         if now - prev > Duration::seconds(STATE_SYNC_TIMEOUT) || error {
-                            download_timeout = true;
+                            download_timeout = now - prev > Duration::seconds(STATE_SYNC_TIMEOUT);
```
Comment (bowenwang1996) [inline chain/client/src/sync.rs]: Why duplicate `now - prev > Duration::seconds(STATE_SYNC_TIMEOUT)`?

File: chain/client/src/sync.rs [hunk 2]
```diff
@@ -696,6 +699,7 @@ impl StateSync {
                 new_shard_sync_download.downloads[0].run_me = false;
                 new_shard_sync_download.downloads[0].state_requests_count += 1;
                 new_shard_sync_download.downloads[0].last_target = Some(target);
+                *update_sync_status = true;
```
Comment (bowenwang1996) [inline chain/client/src/sync.rs]: How is `update_sync_status` used?

File: chain/client/src/sync.rs (hunk 1 above)
Comment (Kouprin) [reply chain/client/src/sync.rs]: why not? if A or B then if A { A happened }

File: chain/client/src/sync.rs (hunk 2 above)
Comment (Kouprin) [reply chain/client/src/sync.rs]: It sets StateSyncResult to Changed or Unchanged. The reason is, the shard state isn't changed if we sent no request.

**Comment by Kouprin** [comment]
Resolved in https://github.com/nearprotocol/nearcore/pull/1904

**Pull request closed without merging**